### ISSUE TITLE
Dangling connections keep node from exiting

### ISSUE THREAD

**Issue by mk-pmb** [ttab/amqp-as-promised#14]
30 seconds after both connections' `.shutdown()` promises have resolved, node still keeps them around:

```text
[WTF Node?] open handles:
- File descriptors: (note: stdio always exists)
  - fd 1 (stdio)
  - fd 2 (stdio)
- Sockets:
  - 127.0.0.1:37554 -> 127.0.0.1:5672
  - 127.0.0.1:37558 -> 127.0.0.1:5672
- Intervals:
  - (30000 ~ 30 s) bound  @ /…/node_modules/amqp-as-promised/node_modules/amqplib/lib/heartbeat.js:65
  - (30000 ~ 30 s) bound  @ /…/node_modules/amqp-as-promised/node_modules/amqplib/lib/heartbeat.js:65
  - (60000 ~ 60 s) bound  @ /…/node_modules/amqp-as-promised/node_modules/amqplib/lib/heartbeat.js:74
  - (60000 ~ 60 s) bound  @ /…/node_modules/amqp-as-promised/node_modules/amqplib/lib/heartbeat.js:74
```

```text
require('amqp-as-promised/package.json').version = 4.2.0
Node.js v9.11.1, npm v5.8.0, Ubuntu 16.04.4 LTS xenial,
kernel: 4.4.0-120-generic, machine/CPU/platform: x86_64 x86_64 x86_64
```

**Comment by mk-pmb** [comment]
It's been annoying for a while now. :-( Do you intend to fix this any time soon?

**Comment by jaymzm** [comment]
Ill take a look at this today and see if I can figure out why it keeps the connections

**Comment by mk-pmb** [comment]
also [Why not to use setInterval()](https://zetafleet.com/blog/2010/04/why-i-consider-setinterval-to-be-harmful.html)

**Comment by jaymzm** [comment]
found some issues with connections and managed to solve the shutdown issue, so should have a new release coming quite soon.

**Comment by jaymzm** [comment]
released a 5.0 version now which should fix the connection issue. the client is now returned as a promise when setup so make sure to check new syntax in the readme.

there was quite a lot of code changes so let me know if you run into any issues with the new version.

**Comment by mk-pmb** [comment]
Seems to work. Thanks! :+1:
